server/src/completion.ts

```diff
@@ -148,10 +148,15 @@ export function asCommitCharacters(kind: ScriptElementKind): string[] | undefine
 export function asResolvedCompletionItem(item: TSCompletionItem, details: tsp.CompletionEntryDetails): TSCompletionItem {
     item.detail = asDetail(details);
     item.documentation = asDocumentation(details);
+    item.deprecated = isDeprecated(details.tags);
```

Comment:
Ironic because `CompletionItem.deprecated` is deprecated itself. The spec says to use `tags` instead:

> @deprecated Use `tags` instead if supported.

Reply:
I understand that supporting `tags` would probably require more work (checking client capability, maybe also updating vscode libraries?) so I'm willing to accept this for now if you think it's too involved for now.

Reply:
Whoops, I didn't notice that since I was just looking at the current dependency chain. I'll look into tags.

Reply:
It looks like it would actually require an update of vscode libraries though so probably should do that in a separate PR first.

Reply:
You've updated the vscode libaries and I've changed to not set the `deprecated` property so resolving.